actions/views/channel.js

```diff
@@ -12,3 +16,11 @@ export function checkAndSetMobileView() {
         });
     };
 }
+
+export function goToChannelById(channelId) {
```

Comment:
I feel like this is not well-suited as an action and it makes it hard to test redux actions with other side-effect (browserHistory.push).
Maybe, create a selector as `getURLByChannelId` and then have the individual component do the `browserHistory.push(props.channelURL)`